### PR TITLE
Centralize metrics used by compactor and add user label to compactor metrics

### PR DESCRIPTION
**Pull request by alexqyle**
**What this PR does**:

Created compactor_metrics.go to have all compactor related metric in one place. Also, added user label to compactor metrics so end user could query metrics per tenant.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/compactor/blocks_cleaner.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ import (
 
 const (
 	defaultDeleteBlocksConcurrency = 16
+	reasonLabelValue               = "retention"
```
Comment (yeya24) [inline pkg/compactor/blocks_cleaner.go:30]: Just curious, the possible values for this are `retention` and `compaction`? What's the main usecase to know whether a block is marked for deletion due to retention or compaction?

And a nit about the naming. Should we call it `reasonValueRetention` instead?

File: pkg/compactor/compactor_metrics.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+const (
+	UserLabelName      = "user"
+	TimeRangeLabelName = "time_range_milliseconds"
+	ReasonLabelName    = "reason"
+)
+
+var (
+	CommonLabels     = []string{UserLabelName}
+	CompactionLabels = []string{TimeRangeLabelName}
+)
```
Comment (yeya24) [inline pkg/compactor/compactor_metrics.go:51]: Nits. Can we keep those variables private? As they are not used by other packages.

**Review by yeya24 — COMMENTED** [review]
Did we check if we changed any existing compactor metrics (names, labels, etc), and it might impact https://github.com/cortexproject/cortex-jsonnet?

Note: If we indeed changed something, should we call out in the changelog?

File: pkg/compactor/blocks_cleaner.go (hunk 1 above)
Comment (alexqyle) [reply pkg/compactor/blocks_cleaner.go:30]: It was there before this change. In my PR, I tried maintaining the original label values. My guess of the purpose of `compaction` label value is that end user tries to know how many blocks were deleted after compaction (at the end of Thanos compact code, source blocks would be marked for deletion if compaction succeeded). The purpose of `retention` label value is the number of blocks marked for deletion during cleaner runtime.

**Comment by alexqyle** [comment]
> Did we check if we changed any existing compactor metrics (names, labels, etc), and it might impact https://github.com/cortexproject/cortex-jsonnet?
> 
> Note: If we indeed changed something, should we call out in the changelog?

There is no metric got renamed in this PR. Some metrics now get extra `user` label. Quickly took a look at `cortex-jsonnet` project. As long as the query is using aggregation, extra label should not result in different result.

**Review by yeya24 — APPROVED** [review]

File: pkg/compactor/syncer_metrics.go [hunk 3]
```diff
@@ -1,124 +0,0 @@
...
-		Help: "Total blocks metadata synchronization failures.",
-	})
-	m.metaSyncDuration = util.NewHistogramDataCollector(prometheus.NewDesc(
-		"cortex_compactor_meta_sync_duration_seconds",
-		"Duration of the blocks metadata synchronization in seconds.",
-		nil, nil))
-	m.metaSyncConsistencyDelay = promauto.With(reg).NewGauge(prometheus.GaugeOpts{
-		Name: "cortex_compactor_meta_sync_consistency_delay_seconds",
-		Help: "Configured consistency delay in seconds.",
-	})
```
Comment (danielblando) [inline pkg/compactor/syncer_metrics.go:49]: I couldn't find this metrics on the new compactor_metrics. Are we removing it?

Comment (alexqyle) [reply pkg/compactor/syncer_metrics.go:49]: Good catch. Will add it to new location

**Pull request merged**